RoyaltiesProcessorDelivery/src/repositories/reservoir_api.ts

```diff
@@ -0,0 +1,193 @@
+import {
+  T_Sale,
+  T_SaleLookupTable,
+  T_Token,
+} from '../types/graphQL_entities_def'
+import { delay } from '../utils/util_functions'
+import { ProjectData } from './subgraph_repository'
+import axios, { AxiosResponse } from 'axios'
+
+require('dotenv').config()
+
+const RESERVOIR_API_KEY = process.env.RESERVOIR_API_KEY
+const MAX_RETRIES = 100
+const SALES_BATCH_SIZE = 500
+const RESERVOIR_TIMEOUT = 5000
+
+type ReservoirSaleEvent = {
+  id: string
+  saleId: string
+  token: {
+    contract: string
+    tokenId: string
+  }
+  orderSource: string
+  orderSide: string
+  orderKind: string
+  from: string
+  to: string
+  amount: number
+  fillSource: string
+  block: number
+  txHash: string
+  timestamp: number
+  price: {
+    currency: {
+      contract: string
+      symbol: string
+    }
+    amount: {
+      raw: string // wei amount
+      decimal: number
+      usd: number
+      native: number
+    }
+  }
+}
+type SalesResponse = {
+  sales: ReservoirSaleEvent[]
+  continuation: string | null
+}
+
+function reservoirSaleModelToSubgraphModel(
+  projectInfo: ProjectData,
+  reservoirSale: ReservoirSaleEvent
+): T_Sale {
+  const _saleType = reservoirSale.amount > 1 ? 'Bundle' : 'Single'
+  // other part of codebase uses length of _summaryTokensSold split by ::
+  // to divide up royalty payments on bundle sales, so use same encoding
+  let _summaryTokensSold = 'dummy'
+
+  // Convert token(s) to array of subgraph's T_SaleLookupTable model
+  const _reservoirLookupTable: T_SaleLookupTable[] = []
+  // populate this with same data as defined in sales_repository.ts
+  if (_saleType === 'Single') {
+    // single sale
+    const _token: T_Token = {
+      id: `${reservoirSale.token.contract}-${reservoirSale.token.tokenId}`,
+      tokenId: parseInt(reservoirSale.token.tokenId),
+      contract: {
+        id: reservoirSale.token.contract,
+      },
+      project: {
+        id: projectInfo.id,
+        name: projectInfo.name ?? '',
+        artistAddress: projectInfo.artistAddress,
+        curationStatus: 'factory', //projectInfo.curationStatus ?? 'factory', // TODO: fix curation status
```

Comment:
can we just remove curationStatus here if we are always using factory?

Reply:
I have it as a TODO to fix - i think this needs to be fixed / remain so that in theory people can query by curation status